server/catgenome/src/test/java/com/epam/catgenome/manager/reference/MotifSearchManagerTestSuccess.java

```diff
@@ -0,0 +1,166 @@
+import com.epam.catgenome.entity.BiologicalDataItemResourceType;
+import com.epam.catgenome.entity.reference.Reference;
+import com.epam.catgenome.entity.reference.motif.Motif;
+import com.epam.catgenome.entity.reference.motif.MotifSearchRequest;
+import com.epam.catgenome.entity.reference.motif.MotifSearchResult;
+import com.epam.catgenome.entity.reference.motif.MotifSearchType;
+import com.epam.catgenome.manager.gene.parser.StrandSerializable;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.core.io.Resource;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import java.io.IOException;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration({"classpath:applicationContext-test.xml"})
+public class MotifSearchManagerTestSuccess {
+
+    @Autowired
+    private MotifSearchManager motifSearchManager;
+
+    @Autowired
+    private ApplicationContext context;
+
+    @Autowired
+    private ReferenceManager referenceManager;
+
+    private static final String A3_FA_PATH = "classpath:templates/A3.fa";
+    private static final String HP_GENOME_PATH = "classpath:templates/reference/hp.genome.fa";
+    private static final String TEST_WG_PATH = "classpath:templates/Test_wg.fa";
+
+    private Resource resource;
+    private Reference referenceOne;
+    private Reference referenceTwo;
+    private long idRef;
+    private long idChr;
+    private long idRefTwo;
+    private long idChrTwo;
+
+    @Before
+    public void registerFileOne() throws IOException {
+        resource = context.getResource(A3_FA_PATH);
+        ReferenceRegistrationRequest request = new ReferenceRegistrationRequest();
+        request.setName("A3.fa");
+        request.setPath(resource.getFile().getPath());
+        request.setType(BiologicalDataItemResourceType.FILE);
+        referenceOne = referenceManager.registerGenome(request);
+        idRef = referenceOne.getId();
+        idChr = referenceOne.getChromosomes().get(0).getId();
+    }
+
+    @Before
+    public void registerFileTwo() throws IOException {
+        resource = context.getResource(HP_GENOME_PATH);
+        ReferenceRegistrationRequest request = new ReferenceRegistrationRequest();
+        request.setName("hp.genome.fa");
+        request.setPath(resource.getFile().getPath());
+        request.setType(BiologicalDataItemResourceType.FILE);
+        referenceTwo = referenceManager.registerGenome(request);
+        idRefTwo = referenceTwo.getId();
+        idChrTwo = referenceTwo.getChromosomes().get(1).getId();
+    }
+
+    @After
+    public void unregisterFiles() throws IOException {
```

Comment:
it is not necessary to do it if you will specify `@Transactional` annotation for this test class, please read about `@Transactional` annotation how this is used and how it is used specifically in tests